### PR TITLE
[Elastic-Agent] Added source uri reloading

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

Adds ability to reload configuration options where restart is not needed (e.g for log level we still need restart)

## Why is it important?

https://github.com/elastic/elastic-agent/issues/678
https://github.com/elastic/kibana/issues/133828

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

**!!! NOT TESTED WITH KIBANA CHANGES YET**

### CODE REVIEW

File: internal/pkg/agent/application/managed_mode.go [hunk 1]
```diff
@@ -207,6 +207,16 @@ func newManaged(
 		agentInfo,
 		cfg,
 		storeSaver,
+		map[string]handlers.ReloadFunc{
```
Comment (michalpristas) [inline internal/pkg/agent/application/managed_mode.go:210]: maybe i will use Reloadables for this

**Comment by criamico** [comment]
Commented in https://github.com/elastic/elastic-agent/pull/686

**Comment by michalpristas** [comment]
@joshdover it could yes, will test mentioned use case and close if ok

**Comment by criamico** [comment]
Thanks for your latest changes @michalpristas. I think that we can test the changes e2e with kibana at this point :)

**Comment by michalpristas** [comment]
@criamico also couple of points for e2e
- when using SNAPSHOT it will use preconfigured URI (not the one provided) as we not only build URI for artifacts but query packages API
- the preconfigured URI on kibana side is incorrect it's `https://artifacts.elastic.co/downloads/elastic-agent` should be `https://artifacts.elastic.co/downloads/` because it's common for all artifacts.

**Comment by michalpristas** [comment]
@joshdover this fixes #408 but only for sourceURI. keeping it open

**Review by AndersonQ — APPROVED** [review]

**Comment by criamico** [comment]
@michalpristas when will the snapshot be available for testing?

**Pull request merged**